### ISSUE TITLE
Separate functions into individual .R files

### ISSUE THREAD

**Issue by MKLau** [ProvTools/Rclean#81]
In the current dev branch there are multiple functions added to the clean.R file. Generally, R packages have separate file. I suppose that this is in mostly convention, but it makes it easier to view the architecture of the package from the directory. This is particularly useful for exported functions. 

Also, this allows each function to have attributions credited to each function individually in the header. 

If there is an explicit reason that necessitates inclusion (i.e. it is necessary for a feature to work), please create a new issue so we can discuss the architectural decisions.

**Comment by MKLau** [comment]
Hi Barb (@blernermhc), 

The documentation for those files can all still end up in the same man page, as @rdname directs  where the documentation occurs (https://cran.r-project.org/web/packages/roxygen2/vignettes/rd.html). They don't need to be in the same .R file to share a .Rd man file. 

At this point, I don't think that this should happen though.

Currently (in dev), clean simply wraps clean.prov in a logic gate to message for the absence of provenance in memory. As it currently exists in the README, the usage of the package is described as loading provenance in memory and using the clean function. This should be the priority and prior to releasing the new functions (such as clean.script), clear and cohesive documentation will need to be added. 

The current structure of the package creates two new functions that are visible to the user in the man files, but the usage of those functions is not made clear to the user at multiple levels of the package documentation, but most importantly at the introduction of the package. This will need to be done prior to release. 

I see a simple solution here, which is to move the 'clean.prov' function's code into the 'clean' function and remove the 'clean.prov' function. Additionally, the 'clean.script' function should be removed from this release and placed onto a separate branch until it's presentation to the user in the context of the package as a whole is added. It seems that the intent here would be to add the 'clean.script' functionality to the 'clean' function, which could/should also be done prior to release.

**Comment by MKLau** [comment]
With regard to authorship, please be sure that you have credited yourself or any of your students that work on functions in the "@author" section. I want everyone who has contributed code to be given appropriate credit, as we have discussed in past conversations regarding other projects. Copyrighting follows these discussions as well, as the initial work for this package was conducted under shared grants.

**Comment by blernermhc** [comment]
I have created a branch, parseR, that contains just the code to allow Rclean to work with the current version of the JSON output, as well as the provParseR and provGraphR packages.  It does not include clean.prov and clean.script.

**Comment by blernermhc** [comment]
I think there is value to the user in having different clean functions that take different types of parameters.  That was the motivation for creating 3 functions.  So, if the user has their provenance in a prov.json option, then they should call clean.  If they have their provenance in a file, they should call clean.prov.  If they do not have provenance, they should call clean.script.  It's really just a convenience for the user.  

In any case, when you talk about "multiple levels of the package documentation" do you mean something other than the man pages and the README?

**Comment by blernermhc** [comment]
The parser_update branch now contains updated documentation:  man pages, README and a new screenshot.  

If you would rather just have Rclean updated to work with the current prov json format, but without clean.prov or clean.script or documentation changes, you can find that on the parseR branch.

**Comment by blernermhc** [comment]
The parser_update branch also contains an updated video demo.
